### PR TITLE
add multidistrict select feature

### PR DESCRIPTION
**Pull request by bennsimon**

### CODE REVIEW

File: opensrp-eusm/src/main/java/org/smartregister/eusm/activity/ProductInfoActivity.java [hunk 1]
```diff
@@ -206,14 +207,14 @@ protected void openLooksGoodConfirmationDialog() {
...
             public void onClick(DialogInterface dialog, int which) {
                 //Do nothing
             }
         });
 
-        builder.setPositiveButton("Looks Good!", new DialogInterface.OnClickListener() {
+        builder.setPositiveButton(getString(R.string.looks_good_button_text), new DialogInterface.OnClickListener() {
             @Override
             public void onClick(DialogInterface dialog, int which) {
                 //TODO add logic to update
```
Comment (Rkareko) [inline opensrp-eusm/src/main/java/org/smartregister/eusm/activity/ProductInfoActivity.java:220]: Is there a ticket to track this TODO?

Comment (bennsimon) [reply opensrp-eusm/src/main/java/org/smartregister/eusm/activity/ProductInfoActivity.java:220]: removed it, thats done.

File: opensrp-eusm/src/main/java/org/smartregister/eusm/layer/SatelliteStreetsLayer.java [hunk 2]
```diff
@@ -31,14 +32,17 @@
...
     private LinkedHashSet<Layer> layers = new LinkedHashSet<>();
+    private String satelliteSourceId = "mapbox://mapbox.satellite";
 
     public SatelliteStreetsLayer(@NonNull Context context) {
         createLayersAndSources(context);
     }
 
     protected void createLayersAndSources(@NonNull Context context) {
-        VectorSource streetSource = new VectorSource(streetSourceId, "mapbox://mapbox.mapbox-streets-v8,mapbox.mapbox-terrain-v2");
+        RasterSource rasterSource = new RasterSource(satelliteSourceId, satelliteSourceId, 256);
```
Comment (Rkareko) [inline opensrp-eusm/src/main/java/org/smartregister/eusm/layer/SatelliteStreetsLayer.java:42]: Refactor `256` to be a  constant

File: opensrp-eusm/src/main/java/org/smartregister/eusm/layer/SatelliteStreetsLayer.java [hunk 3]
```diff
@@ -31,14 +32,17 @@
...
+        VectorSource streetSource = new VectorSource(streetSourceId, "mapbox://mapbox.mapbox-streets-v8");
```
Comment (Rkareko) [inline opensrp-eusm/src/main/java/org/smartregister/eusm/layer/SatelliteStreetsLayer.java:43]: Refactor `mapbox://mapbox.mapbox-streets-v8` to be a constant

File: opensrp-eusm/src/main/java/org/smartregister/eusm/presenter/EusmBaseDrawerPresenter.java [hunk 4]
```diff
@@ -88,15 +88,33 @@ public String getEntireTree(List<FormLocation> entireTree) {
...
-
+        cleanUpFormLocations(formLocations);
         return AssetHandler.javaToJsonString(formLocations,
                 new TypeToken<List<FormLocation>>() {
                 }.getType());
     }
 
+
+    private void cleanUpFormLocations(List<FormLocation> formLocations) {
+        if (!formLocations.isEmpty()) {
```
Comment (Rkareko) [inline opensrp-eusm/src/main/java/org/smartregister/eusm/presenter/EusmBaseDrawerPresenter.java:103]: we may require a null check for `formLocations`

File: opensrp-eusm/src/main/java/org/smartregister/eusm/util/AppUtils.java [hunk 5]
```diff
@@ -203,6 +203,12 @@ public static String getStringFromJsonElement(JsonObject jsonObject, String key)
         });
     }
 
+    public static Set<org.smartregister.domain.Location> getOperationalAreaLocations(Set<String> operationalAreas) {
+        return EusmApplication.getInstance().getAppLocationRepository()
+                .getLocationByNameAndGeoLevel(operationalAreas, "2");//restrict to district geographic level
```
Comment (Rkareko) [inline opensrp-eusm/src/main/java/org/smartregister/eusm/util/AppUtils.java:208]: refactor `2` to a constant

**Review by Rkareko — APPROVED** [review]

**Pull request merged**